test/LogTest.php

```diff
@@ -0,0 +1,222 @@
+
+        $content = $this->readLog();
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetLevelToNullHavingCustomLevels()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->addPriority('TEST', 8);
+        $opusLog->setLevel(null);
+        $opusLog->addPriority('TLEVEL', 9);
+
+        $testMessage = 'Test level message';
+        $opusLog->test($testMessage);
+
+        $tlevelMessage = 'Tlevel message';
+        $opusLog->tlevel($tlevelMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($testMessage, $content);
+        $this->assertContains($tlevelMessage, $content);
+    }
+
+    public function testSetLevelNotEffectingOtherFilters()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $filter = new \Zend_Log_Filter_Priority(\Zend_Log::WARN);
+        $opusLog->addFilter($filter);
+
+        $infoMessage = 'Info level Message';
+        $opusLog->info($infoMessage);
+
+        $opusLog->setLevel(\Zend_Log::NOTICE);
+        $noticeMessage = 'Notice Level Message';
+        $opusLog->notice($noticeMessage);
+
+        $opusLog->setLevel(\Zend_Log::ERR);
+        $errorMessage = 'Error level Message';
+        $opusLog->err($errorMessage);
+
+        $content = $this->readLog();
+
+        $this->assertNotContains($infoMessage, $content);
+        $this->assertNotContains($noticeMessage, $content);
+        $this->assertContains($errorMessage, $content);
+    }
+
+    public function testSetLevelNegativeLevel()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $opusLog->setLevel(-1);
+    }
+
+    public function testSetLevelStringArgument()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $opusLog->setLevel('TestLevel');
+    }
+
+    public function testSetLevelStringLevel()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $opusLog->setLevel('7');
+
+        $this->assertEquals(\Zend_Log::DEBUG, $opusLog->getLevel());
+    }
+
+    public function testGetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $level = $opusLog->getLevel();
+
+        $this->assertEquals(\Zend_Log::INFO, $level);
+    }
+
+    public function testGetLevelReturnsNull()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+        $level = $opusLog->getLevel();
+
+        $this->assertNull($level);
+    }
+
+    public function testGet()
+    {
+        \Zend_Registry::set('Zend_Log', $this->getOpusLog());
+        $log = Log::get();
+
+        $this->assertSame(\Zend_Registry::get('Zend_Log'), $log);
+    }
+
+    public function testDrop()
```

Comment:
This works? Why? I mean `get` just gets the logger from `Zend_Registry`, so here we should get the same logger twice. If it isn't the same object, either the assertion works differently than I think or `Zend_Registry`. Is the registry creating clones of objects before returning them?

Reply:
Any thoughts here? Without comments it is hard to tell if an issue has been addressed. It doesn't look like it.